wheatley/bot.py

```diff
@@ -219,15 +236,16 @@ def start_method(self):
         Called when the ringing is about to go into changes.
         Resets the row_generator and starts the next row.
         """
-        assert self.row_generator.number_of_bells == self._tower.number_of_bells, \
-            f"{self.row_generator.number_of_bells} != {self._tower.number_of_bells}"
-
-        self.row_generator.reset()
-        self.start_next_row()
+        if self._check_number_of_bells():
+            self.row_generator.reset()
+            self.start_next_row()
 
     def tick(self):
         """ Called every time the main loop is executed when the bot is ringing. """
-        bell = Bell.from_index(self._place) if self._is_ringing_rounds else self._row[self._place]
+        if self._is_ringing_rounds or self._place >= len(self._row):
```

Comment:
Perhaps a comment here to just make it clear what this `if` is doing?  I guess the code is fairly self explanatory, but it took me about 30 seconds to work out exactly what it did and a comment would make reading the code much smoother.